packages/pwa/app/utils/url.js

```diff
@@ -185,3 +195,66 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path with locale/site value to the path as url path or url query param
+ * based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses?locale=en-GB
+ *
+ *  @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "path"
+ * }
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /en-GB/women/dresses
+ *
+ */
+export const buildPathWithUrlConfig = (url, configValues = {}) => {
```

Comment:
It would be nice to see a test specifically for this function.

Reply:
Yes, I am in the process of adding and fixing tests on this PR